Cubical/Structures/Monoid.agda

```diff
@@ -43,6 +43,47 @@ monoid-structure = add-to-structure (raw-monoid-structure) monoid-axioms
 Monoids : Type (ℓ-suc ℓ)
 Monoids {ℓ} = TypeWithStr ℓ monoid-structure
 
+-- Monoid extractors
+
+⟨_⟩ : Monoids {ℓ} → Type ℓ
+⟨ G , _ ⟩ = G
+
+monoid-id : (M : Monoids {ℓ}) → ⟨ M ⟩
+monoid-id (_ , (e , _) , _) = e
+
+monoid-operation : (M : Monoids {ℓ}) → ⟨ M ⟩ → ⟨ M ⟩ → ⟨ M ⟩
+monoid-operation (_ , (_ , f) , _) = f
+
+module monoid-syntax where
+  id : (M : Monoids {ℓ}) → ⟨ M ⟩
+  id = monoid-id
+
+  monoid-operation-syntax : (M : Monoids {ℓ}) → ⟨ M ⟩ → ⟨ M ⟩ → ⟨ M ⟩
+  monoid-operation-syntax = monoid-operation
+
+  infixr 20 monoid-operation-syntax
+  syntax monoid-operation-syntax M x y = x ·⟨ M ⟩ y
```

Comment:
Quick question: would it be possible to use instance arguments to automatically infer `M`? @Saizan

Reply:
Today I've read the documentation about instances and I'm note sure if this is what you want but since records and modules are really related in agda (I think I've read that records are modules or modules are records, but don't trust me on this) you can use the modules multiplicative-notation and additive-notation in the next style:
open import multiplicative-notation {{...}} 
and the argument (G : Group) becomes an instance argument.

Reply:
I see. So we won't have to write it explicitly if it's clear from the context which group we are referring to?

Reply:
Well, if you have declare an instance and the groups have distinct domain I´m sure it works, I have tried it with multiplicative-notation from Group. But if you want the same syntax for monoids and all the structures I have to define it yet, I can write the code tonight.

Reply:
With this I'm having two different alternatives:
1. Different structures can't have the same notation at the same time, so if we have _\cdot_ for groups in one file then we can't have _\cdot_ for monoids also in the same file.
2. Every structure have every notation. So if we have a group and a monoid in the same file, and we import the _+_ notation for the Group and _\cdot_ notation for the monoid nothing will stop of us typing x + y where x and y are elements of the group (Note that the same is true for 2 distincts groups in the option 1).
For me the ideal will be a mixture but at the moment I don't know how to implement it.
Any preferences between the above 2 options?

Reply:
Actually I think I got a solution, sorry for the ping.